### DISCUSSION TITLE
Row ID / selected row

### DISCUSSION THREAD

**Discussion by Dee66** [nick-keller/react-datasheet-grid#210] [Q&A] [answered]
First off, great component, MUCH better than most options out there.

Is there any way of adding event handling on row click? I want to load a new page based on the id of the selected / clicked row. Almost like stickyRightColumn but I can't for the life of me work out how to pass the selected row to a custom handler.

I can get the id if the data is edited first...but how do I get the ID (or whole row) on 'rowClick'?

**Comment by nick-keller** [comment, marked as answer]
Hey @Dee66 ,
1. You can add a key `selected: true` to the row object itself. You can then use it to add custom class to the row, show a checkbox in the gutter, you name it...
2. You can also use the [onActiveCellChange](https://react-datasheet-grid.netlify.app/docs/api-reference/props#onactivecellchange) prop to know when the active cell (the highlighted cell) changes

**Comment by Dee66** [comment]
That's perfect!

onActiveCellChange gives me the row ID, I can use that to do a lookup on
the grid data and trigger the page load.

Can't say I've got 'selected' to work but the cell change works a treat.

Thanks bro 👌
